checkpoint/recovery_test.go

```diff
@@ -195,24 +145,30 @@ func TestRecoverFromHttp(t *testing.T) {
 
 			fs := afero.NewMemMapFs()
 			cfg := &checkpoint.RecoverConfig{
-				GoldenAtx:      types.ATXID{1},
+				GoldenAtx:      goldenAtx,
+				PostDataDir:    t.TempDir(),
 				DataDir:        t.TempDir(),
 				DbFile:         "test.sql",
 				PreserveOwnAtx: true,
+				NodeID:         types.NodeID{2, 3, 4},
+				Uri:            tc.uri,
+				Restore:        types.LayerID(recoverLayer),
 			}
 			bsdir := filepath.Join(cfg.DataDir, bootstrap.DirName)
 			require.NoError(t, fs.MkdirAll(bsdir, 0o700))
 			db := sql.InMemory()
-			newdb, err := checkpoint.RecoverWithDb(ctx, logtest.New(t), db, fs, cfg, types.NodeID{2, 3, 4}, tc.uri, types.LayerID(recoverLayer))
-			if len(tc.expErr) > 0 {
-				require.ErrorContains(t, err, tc.expErr)
+			preserve, err := checkpoint.RecoverWithDb(ctx, logtest.New(t), db, fs, cfg)
+			if tc.expErr != nil {
+				require.ErrorIs(t, err, tc.expErr)
 				return
 			}
 			require.NoError(t, err)
```

Comment:
No need to be careful. `errors.Is` can handle `nil`:

```suggestion
			require.ErrorIs(t, err, tc.expErr)
```

https://go.dev/play/p/yZfoyOh7n2g

Reply:
i did this intentionally because i am returning immediately when there is error,